sc2/bot_ai.py

```diff
@@ -370,34 +385,48 @@ def can_afford(
             ):  # cant replace 1 with "Target.None.value" because ".None" doesnt seem to be a valid enum name
                 return True
             # Check if able to use ability on a unit
-            elif (
+            if (
                 ability_target in {Target.Unit.value, Target.PointOrUnit.value}
                 and isinstance(target, Unit)
                 and unit.position._distance_squared(target.position) <= cast_range ** 2
             ):
                 return True
             # Check if able to use ability on a position
-            elif (
+            if (
                 ability_target in {Target.Point.value, Target.PointOrUnit.value}
                 and isinstance(target, (Point2, Point3))
                 and unit.position._distance_squared(target) <= cast_range ** 2
             ):
                 return True
         return False
 
-    def select_build_worker(self, pos: Union[Unit, Point2, Point3], force: bool=False) -> Optional[Unit]:
-        """Select a worker to build a building with."""
-        workers = self.workers.filter(lambda w: (w.is_gathering or w.is_idle) and w.distance_to(pos) < 20) or self.workers
+    def select_build_worker(self, pos: Union[Unit, Point2, Point3], force: bool = False) -> Optional[Unit]:
+        """Select a worker to build a building with.
+         Prioritizes closest workers to given position that are not carrying any resource or are idle
+        Example usage:
+        self.actions = []
+        selected_worker = self.select_build_worker(given_position)
+        if selected_worker:
+            self.action.append(selected_worker.build(building, given_position))
+        if actions:
+            await self.do_actions(self.actions)"""
```

Comment:
...and same comment as for `get_next_expansion()` Looking at the example in isolation, the use of the `self.actions` list is a bit contrived.